app/services/deployment/configuration_builders/by_project.rb

```diff
@@ -103,7 +108,8 @@ def build_repo_configuration_by_project(repository, branches)
           repo_path: repository.path,
           git_reference: branches.fetch(repository.name, "master"),
           project_integration_id: @project.integration_id,
-          project_integration_type: @project.integration_type
+          project_integration_type: @project.integration_type,
+          repo_settings: PROVIDER_REPOSITORY_SETTINGS.fetch(@project.integration_type).call(repository)
```

Comment:
Not sure that this is a good idea. I have to remember that in one case this is a deploy token in another case this is an integration_id. All this can confuse future developers.

BTW
```
    # Temporary solution. <---    your comment
    def add_private_key(private_key)
      return self unless repo_configuration.project_integration_type == ProjectsConstants::Providers::VIA_SSH

      repo_configuration.project_integration_id = private_key
      self
    end
```